macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,262 @@
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header:
+        pos: param2
+        type: sound_header
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+  sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_samples
+        -orig-id: length
+        type: u4
+        doc: number of samples
+        if: sound_header_type == sound_header_type::standard
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+        if: sound_header_type != sound_header_type::standard
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: midi_note
+        type: u1
+        doc: base frequency of sample, expressed as MIDI note values, 60 is middle C
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+        if: sound_header_type != sound_header_type::standard
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+        if: sound_header_type != sound_header_type::standard
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+        if: sound_header_type != sound_header_type::standard
+      - id: instrument_chunk_ptr
+        -orig-id: instrumentChunks
+        type: u4
+        doc: pointer to instrument info
+        if: sound_header_type == sound_header_type::extended
+      - id: aes_recording_ptr
+        -orig-id: AESRecording
+        type: u4
+        doc: pointer to audio info
+        if: sound_header_type == sound_header_type::extended
+      - id: format
+        size: 4
+        type: str
+        encoding: ASCII
+        doc: data format type
+        if: sound_header_type == sound_header_type::compressed
+      - id: reserved
+        -orig-id: futureUse2
+        size: 4
+        if: sound_header_type == sound_header_type::compressed
+      - id: state_vars_ptr
+        -orig-id: stateVars
+        type: u4
+        doc: pointer to StateBlock
+        if: sound_header_type == sound_header_type::compressed
```

Comment:
You got the point about reusing code, but I'd prefer extracting these fields from the `compressed_sound_header` to a special subtype. There are a lot of consecutive `if: sound_header_type == sound_header_type::compressed` to my taste.

Again, I refer to the [`bmp.ksy` spec](https://github.com/kaitai-io/kaitai_struct_formats/blob/d1e58e36f26ecff1375c6022dc040342914ee4d1/image/bmp.ksy#L350-L372), where I solved this by splitting the `bitmap_header` to multiple extensions _(subtypes), which are enabled/disabled as a whole using only one `if` per extension. IMHO this makes the spec clearer and more understandable.

Reply:
I'll have a look 👀